### PR TITLE
planner: simplify global and session binding handle

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #48875

Problem Summary: planner: simplify global and session binding handle

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/bindinfo/global_handle.go [hunk 1]
```diff
@@ -486,6 +468,9 @@ func (tmpMap *tmpBindRecordMap) Add(bindRecord *BindRecord) {
 
 // DropInvalidGlobalBinding executes the drop BindRecord tasks.
 func (h *globalBindingHandle) DropInvalidGlobalBinding() {
+	defer func() {
+		_ = h.LoadFromStorageToCache(false)
```
Comment (hawkingrei) [inline pkg/bindinfo/global_handle.go:472]: We can add a warn log here.

**Review by elsa0520 — APPROVED** [review]
LGTM

File: pkg/infoschema/test/clustertablestest/tables_test.go [hunk 2]
```diff
@@ -1011,39 +1011,6 @@ func TestStmtSummaryInternalQuery(t *testing.T) {
...
-
-	// Create a new session to test.
-	tk = newTestKitWithRoot(t, store)
-
-	tk.MustExec("select * from t where t.a = 1")
-	tk.MustQuery(`select exec_count, digest_text
-		from information_schema.statements_summary
-		where digest_text like "select original_sql , bind_sql , default_db , status%"`).Check(testkit.Rows())
-
-	// Enable internal query and evolve baseline.
```
Comment (qw4990) [inline pkg/infoschema/test/clustertablestest/tables_test.go:1025]: Plan evolution is deprecated.

**Review by hawkingrei — APPROVED** [review]

**Review by ywqzzy — APPROVED** [review]

**Pull request merged**